### PR TITLE
CB-2638: Oracle wallet authentication support. Read features

### PR DESCRIPTION
**Pull request by oksana-cherniavskaia**

### CODE REVIEW

File: plugins/org.jkiss.dbeaver.model/src/org/jkiss/dbeaver/runtime/properties/ObjectPropertyDescriptor.java [hunk 1]
```diff
@@ -222,10 +220,21 @@ public boolean isEditable(Object object)
         return getEditableValue(object);
     }
 
+    private List<String> parseFeatures() {
+        String features = propInfo.features();
+        if (features == null || features.isBlank()) {
```
Comment (LonwoLonwo) [inline plugins/org.jkiss.dbeaver.model/src/org/jkiss/dbeaver/runtime/properties/ObjectPropertyDescriptor.java]: Please look at the CommonUtils#isNotEmpty method in the future

File: plugins/org.jkiss.dbeaver.model/src/org/jkiss/dbeaver/runtime/properties/ObjectPropertyDescriptor.java [hunk 2]
```diff
@@ -243,11 +252,19 @@ public String[] getFeatures() {
...
         if (this.isViewable()) features.add(DBConstants.PROP_FEATURE_VIEWABLE);
         if (this.isPassword()) features.add(DBConstants.PROP_FEATURE_PASSWORD);
+
         return features.toArray(new String[0]);
     }
 
     @Override
     public boolean hasFeature(@NotNull String feature) {
+
+        List<String> features = parseFeatures();
```
Comment (serge-rider) [inline plugins/org.jkiss.dbeaver.model/src/org/jkiss/dbeaver/runtime/properties/ObjectPropertyDescriptor.java]: Function hasFeature may be called 1000 times per second. It is not good to parse, create array, etc for every call.

File: plugins/org.jkiss.dbeaver.model/src/org/jkiss/dbeaver/runtime/properties/ObjectPropertyDescriptor.java (hunk 1 above)
Comment (serge-rider) [reply plugins/org.jkiss.dbeaver.model/src/org/jkiss/dbeaver/runtime/properties/ObjectPropertyDescriptor.java]: Or maybe we should make it String[] features() default {};

**Review by serge-rider — CHANGES_REQUESTED** [review]

File: plugins/org.jkiss.dbeaver.model/src/org/jkiss/dbeaver/runtime/properties/ObjectPropertyDescriptor.java (hunk 1 above)
Comment (oksana-cherniavskaia) [reply plugins/org.jkiss.dbeaver.model/src/org/jkiss/dbeaver/runtime/properties/ObjectPropertyDescriptor.java]: done

File: plugins/org.jkiss.dbeaver.model/src/org/jkiss/dbeaver/runtime/properties/ObjectPropertyDescriptor.java [hunk 3]
```diff
@@ -280,7 +283,10 @@ public boolean hasFeature(@NotNull String feature) {
             case DBConstants.PROP_FEATURE_PASSWORD:
                 return this.isPassword();
         }
-        return false;
+
+        String[] features = propInfo.features();
```
Comment (serge-rider) [inline plugins/org.jkiss.dbeaver.model/src/org/jkiss/dbeaver/runtime/properties/ObjectPropertyDescriptor.java]: Again, this function may be called 10000 times per second. Do not use expensive operations including streams.
Use check for empty array (99% cases) + ArrayUtils.contains.

File: plugins/org.jkiss.dbeaver.model/src/org/jkiss/dbeaver/runtime/properties/ObjectPropertyDescriptor.java [hunk 4]
```diff
@@ -280,7 +284,12 @@ public boolean hasFeature(@NotNull String feature) {
             case DBConstants.PROP_FEATURE_PASSWORD:
                 return this.isPassword();
         }
-        return false;
+
+        if (ArrayUtils.isEmpty(propInfo.features())) {
```
Comment (alexander-skoblikov) [inline plugins/org.jkiss.dbeaver.model/src/org/jkiss/dbeaver/runtime/properties/ObjectPropertyDescriptor.java]: such  check is already exist in ArrayUtils.contains

File: plugins/org.jkiss.dbeaver.model/OSGI-INF/l10n/bundle.properties [hunk 5]
```diff
@@ -156,6 +156,8 @@ meta.org.jkiss.dbeaver.model.navigator.DBNResource.resourceSize.name=Size
 meta.org.jkiss.dbeaver.model.impl.auth.AuthModelDatabaseNativeCredentials.userName.name = User name
 meta.org.jkiss.dbeaver.model.impl.auth.AuthModelDatabaseNativeCredentials.userPassword.name = User password
 
+meta.com.dbeaver.db.oracle.net.auth.wallet.OracleAuthModelWalletCredentials.walletArchive.name = Oracle wallet archive
```
Comment (serge-rider) [inline plugins/org.jkiss.dbeaver.model/OSGI-INF/l10n/bundle.properties]: This must be in Oracle EE bundle

**Review by serge-rider — CHANGES_REQUESTED** [review]

File: plugins/org.jkiss.dbeaver.model/OSGI-INF/l10n/bundle.properties (hunk 5 above)
Comment (oksana-cherniavskaia) [reply plugins/org.jkiss.dbeaver.model/OSGI-INF/l10n/bundle.properties]: done

File: plugins/org.jkiss.dbeaver.model/src/org/jkiss/dbeaver/runtime/properties/ObjectPropertyDescriptor.java (hunk 4 above)
Comment (oksana-cherniavskaia) [reply plugins/org.jkiss.dbeaver.model/src/org/jkiss/dbeaver/runtime/properties/ObjectPropertyDescriptor.java]: addressed

**Review by serge-rider — APPROVED** [review]

**Pull request merged**